src/components/Hoverable/index.js

```diff
@@ -22,11 +22,11 @@ class Hoverable extends Component {
      */
     toggleHoverState() {
         if (this.state.isHovered) {
-            this.props.onHoverOut();
             this.setState({isHovered: false});
+            this.props.onHoverOut();
```

Comment:
This order had to change because if `this.props.onHoverOut` triggers a re-render then it's not guaranteed that `this.state.isHovered` will have updated. This way, we update the internal state _then_ do any side-effects of the `onHoverOut` callback.